pkg/api/util/names.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+)
+
+func ExpectedCertificateRequestName(crt *cmapi.Certificate) (string, error) {
```

Comment:
Given this is being made part of the api/util package, which should generally be something we avoid changing (although not as a *hard* rule), can we consider the name of this function better? Perhaps `ComputeCertificateRequestName`?